<a href="https://colab.research.google.com/github/vithika-karan/Email-Marketing-Effectiveness-Prediction/blob/main/Email_Campaign_Effectiveness_Prediction_Vithika_Karan.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Project Title : Email Campaign Effectiveness Prediction**


##**Problem Description**

Most of the small to medium business owners are making effective use of Gmail-based
Email marketing Strategies for offline targeting of converting their prospective customers into
leads so that they stay with them in Business.
The main objective is to create a machine learning model to characterize the mail and track
the mail that is ignored; read; acknowledged by the reader.
Data columns are self-explanatory.

##Introduction:
Email Marketing can be defined as a marketing technique in which businesses stay connected with their customers through emails, making them aware about their new products, updates, important notices related to the products they are using.

Most importantly, email marketing allows businesses to build relationships with leads, customers and past customers. It's a way to communicate directly to the customers in their inbox, at a time that is convenient for them. With the right messaging tone and strategies, emails are one of the most important marketing channels.

We all subscribe to many different kinds of businesses through emails because it's required to do so, sometimes to get digital receipts of the things we bought or to get digital information about the business to stay updated. But many of times we do not tend to read an email due to a number of reasons - to name a few would be no proper structure, too many images, too many links inside the mail, complex vocabulary used or simply too long emails.

In this problem statement, we will be trying to create machine learning models that characterize and predict whether the mail is ignored, read or acknowledged by the reader. In addition to this, we will be trying to analyze and find all the features that are important for an email to not get ignored.

So let's begin!

In [1]:
#Importing important libraries and modules
#for data reading and manipulation
import pandas as pd
import numpy as np

#for data visualization
import matplotlib as mpl
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
plt.rcParams.update({'figure.figsize':(7,4),'figure.dpi':100})

#to ignore warnings
import warnings    
warnings.filterwarnings('ignore')

In [2]:
#reading the csv dataset
df_orig = pd.read_csv("/content/drive/MyDrive/ALMABETTER/CAPSTONE PROJECTS/Email Campaign Effectiveness Prediction - Vithika Karan/Copy of data_email_campaign.csv")

In [3]:
#first look of our dataset
df_orig.head()

Email_ID  Email_Type  ...  Total_Images  Email_Status
0  EMA00081000034500           1  ...           0.0             0
1  EMA00081000045360           2  ...           0.0             0
2  EMA00081000066290           2  ...           0.0             1
3  EMA00081000076560           1  ...           0.0             0
4  EMA00081000109720           1  ...           0.0             0

[5 rows x 12 columns]

In [4]:
#basic info about our data
df_orig.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 68353 entries, 0 to 68352
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Email_ID                   68353 non-null  object 
 1   Email_Type                 68353 non-null  int64  
 2   Subject_Hotness_Score      68353 non-null  float64
 3   Email_Source_Type          68353 non-null  int64  
 4   Customer_Location          56758 non-null  object 
 5   Email_Campaign_Type        68353 non-null  int64  
 6   Total_Past_Communications  61528 non-null  float64
 7   Time_Email_sent_Category   68353 non-null  int64  
 8   Word_Count                 68353 non-null  int64  
 9   Total_Links                66152 non-null  float64
 10  Total_Images               66676 non-null  float64
 11  Email_Status               68353 non-null  int64  
dtypes: float64(4), int64(6), object(2)
memory usage: 6.3+ MB


##Data Summary:
Our email campaign dataset have 68353 observations and 12 features. Clearly Email_Status is our target variable.

Our features:
* Email Id - It contains the email id's of the customers/individuals
* Email Type - There are two categories 1 and 2. We can think of them as marketing emails or important updates, notices like email regarding the businesses.
* Subject Hotness Score - It is the email's subject's score on the basis of how good and effective the content is
* Email Source - It represents the source of the email like sales and marketing or important admin mails related to the product
* Email Campaign Type - The campaign type
* Total Past Coomunications - This column contains the total previous mails from the same source, the number of communications had
* Customer Location - Contains demographical data of the customer, the location where the customer resides
* Time Email sent Category - It has three categories 1,2 and 3; the time of the day when the email was sent, we can think of it as morning, evening and night time slots.
* Word Count - The number of words contained in the email
* Total links - Number of links in the email
* Total Images - Number of images in the email
* Email Status - Our target variable which contains whether the mail was ignored, read, acknowledged by the reader